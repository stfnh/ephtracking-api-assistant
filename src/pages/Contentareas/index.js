import React, { Fragment } from 'react';

import CodeBlock from '../../components/CodeBlock';
import Preview from '../../components/Preview';

const Contentareas = () => {
  // TODO: Move endpoints (API calls) to file as constant and import where needed (easier to maintain)
  const url = `https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/json`;
  return (
    <Fragment>
      <h1 className="title">Retrieving List of all Content Areas</h1>
      <h5 className="title is-5">Usage</h5>
      <CodeBlock>
        https://ephtracking.cdc.gov/apigateway/api/{'{'}version{'}'}/contentareas/{'{'}getChildMeasure{'}'}/{'{'}getMultiMeasure{'}'}/{'{'}returnType{'}'}[?apiToken]
      </CodeBlock>
      <hr />
      <Preview url={url} />
    </Fragment>
  );
};

export default Contentareas;
