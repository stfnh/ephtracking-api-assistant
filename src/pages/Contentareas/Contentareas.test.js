import React from 'react';
import { shallow } from 'enzyme';

import Contentareas from './';

describe('Page Contentareas', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Contentareas />);
    expect(wrapper).toMatchSnapshot();
  });
});
