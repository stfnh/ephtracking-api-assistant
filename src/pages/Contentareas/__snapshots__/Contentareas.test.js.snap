// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Contentareas renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contentareas />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Retrieving List of all Content Areas
</h1>,
        <h5
          className="title is-5"
>
          Usage
</h5>,
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/
          {
          version
          }
          /contentareas/
          {
          getChildMeasure
          }
          /
          {
          getMultiMeasure
          }
          /
          {
          returnType
          }
          [?apiToken]
</CodeBlock>,
        <hr />,
        <Preview
          url="https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/json"
          validUrl={true}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retrieving List of all Content Areas",
          "className": "title",
        },
        "ref": null,
        "rendered": "Retrieving List of all Content Areas",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Usage",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Usage",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/contentareas/",
            "{",
            "getChildMeasure",
            "}",
            "/",
            "{",
            "getMultiMeasure",
            "}",
            "/",
            "{",
            "returnType",
            "}",
            "[?apiToken]",
          ],
        },
        "ref": null,
        "rendered": Array [
          "https://ephtracking.cdc.gov/apigateway/api/",
          "{",
          "version",
          "}",
          "/contentareas/",
          "{",
          "getChildMeasure",
          "}",
          "/",
          "{",
          "getMultiMeasure",
          "}",
          "/",
          "{",
          "returnType",
          "}",
          "[?apiToken]",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/json",
          "validUrl": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Retrieving List of all Content Areas
</h1>,
          <h5
            className="title is-5"
>
            Usage
</h5>,
          <CodeBlock>
            https://ephtracking.cdc.gov/apigateway/api/
            {
            version
            }
            /contentareas/
            {
            getChildMeasure
            }
            /
            {
            getMultiMeasure
            }
            /
            {
            returnType
            }
            [?apiToken]
</CodeBlock>,
          <hr />,
          <Preview
            url="https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/json"
            validUrl={true}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retrieving List of all Content Areas",
            "className": "title",
          },
          "ref": null,
          "rendered": "Retrieving List of all Content Areas",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Usage",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Usage",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "https://ephtracking.cdc.gov/apigateway/api/",
              "{",
              "version",
              "}",
              "/contentareas/",
              "{",
              "getChildMeasure",
              "}",
              "/",
              "{",
              "getMultiMeasure",
              "}",
              "/",
              "{",
              "returnType",
              "}",
              "[?apiToken]",
            ],
          },
          "ref": null,
          "rendered": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/contentareas/",
            "{",
            "getChildMeasure",
            "}",
            "/",
            "{",
            "getMultiMeasure",
            "}",
            "/",
            "{",
            "returnType",
            "}",
            "[?apiToken]",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "https://ephtracking.cdc.gov/apigateway/api/v1/contentareas/json",
            "validUrl": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
